core/src/main/java/org/kiwix/kiwixmobile/core/webserver/ZimHostActivity.java

```diff
@@ -305,4 +296,27 @@ private void launchTetheringSettingsScreen() {
       Toast.LENGTH_SHORT)
       .show();
   }
+
+  private void selectPreviouslyHostedBooks(@Nullable List<BooksOnDiskListItem> books) {
+    selectedBooksId.addAll(sharedPreferenceUtil.getHostedBooks());
+    if (books != null && !books.isEmpty()) {
+      if (selectedBooksId.isEmpty()) {
+        // Select all books if no book ids are stored
+        for (BooksOnDiskListItem book : books) {
+          if (book instanceof BooksOnDiskListItem.BookOnDisk) {
+            selectedBooksId.add(((BooksOnDiskListItem.BookOnDisk) book).getBook().getId());
+            book.setSelected(true);
```

Comment:
this should be made intrinsic, you can write an intialiser in BookOnDisk that makes this the case

Reply:
Actually this was misguided advice, the BookOnDisks are used in other contexts and we don't want them by default selected. The presenter might be better suited to feeding us a list of books with all selected though.